src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor.cs

```diff
@@ -296,4 +276,88 @@ public void Dispose()
         _logsSubscription?.Dispose();
         _filterCts?.Dispose();
     }
+
+    public UrlState GetUrlFromSerializableViewModel(StructuredLogsPageState serializable)
+    {
+        var path = serializable.SelectedApplication is not null ? $"/StructuredLogs/{serializable.SelectedApplication}" : "/StructuredLogs";
+        var queryParameters = new Dictionary<string, string?>();
+
+        if (serializable.LogLevelText is not null)
+        {
+            queryParameters.Add("level", serializable.LogLevelText);
+        }
+
+        if (serializable.Filters.Count > 0)
+        {
+            queryParameters.Add("filters", JsonSerializer.Serialize(serializable.Filters));
```

Comment:
I'm happy to do either. What do you mean by not liking the service instance id guid?

Reply:
This:

![image](https://github.com/dotnet/aspire/assets/303201/4c7c4838-67bb-4e1e-85bd-25e767e45ebe)

Console logs does it better:

![image](https://github.com/dotnet/aspire/assets/303201/f40ca9a6-0111-45d0-9dca-6aa46f5a5938)

Reply:
For now, I made a simple filter parser

Similar to github, + separates individual filters, and to avoid confusion, there's a second parameter (separated by a colon) signifying the condition. The filter value is url encoded

ie:

`filters=Message:contains:app+Application:!equals:frontend`

Reply:
re: your comment about service instance ID, I see that's used in other pages too. I'll change all of them in a separate PR and request your review

Reply:
Yes, colon will be url-encoded